spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/health/PubSubHealthIndicatorAutoConfiguration.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.pubsub.health;
+
+import org.springframework.boot.actuate.autoconfigure.health.ConditionalOnEnabledHealthIndicator;
+import org.springframework.boot.actuate.autoconfigure.health.HealthContributorAutoConfiguration;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cloud.gcp.autoconfigure.pubsub.GcpPubSubAutoConfiguration;
+import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * {@link HealthContributorAutoConfiguration Auto-configuration} for
+ * {@link PubSubHealthIndicator}.
+ * 
+ * @author Vinicius Carvalho
+ *
+ * @since 1.3
+ */
+@Configuration
+@ConditionalOnClass(HealthIndicator.class)
+@ConditionalOnEnabledHealthIndicator("pubsub")
+@AutoConfigureBefore(HealthContributorAutoConfiguration.class)
+@AutoConfigureAfter(GcpPubSubAutoConfiguration.class)
+public class PubSubHealthIndicatorAutoConfiguration {
```

Comment:
Why is this one still top-level class?
What is wrong with Spring Boot out-of-the-box approaches with inner configuration classes?
This way you don't need to add a new `spring.factories` entry

Reply:
Artem we spoke about this, and I believe the intention was to put a separate PR to move this one and probably the Datastore into inner configuration

Reply:
Sorry if I missed (or forgot) that somehow: too many on my plate.

OK then. Not sure why don't do that right now for this specific class: looks like everything is independent.
But would be great if you we have a dedicated issue do not forgot to do that.

Thanks and sorry again.

Reply:
Hey @artembilan no worries man, notneed to apologize, thanks for you having such eye for detail and keeping the quality so high. We just thought on doing the refactoring in one batch with all other auto configuration classes :)